src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseJunctions.FileSystem.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public abstract class BaseJunctions_FileSystem : BaseSymbolicLinks
+    {
+        protected DirectoryInfo CreateJunction(string junctionPath, string targetPath)
+        {
+            Assert.True(MountHelper.CreateJunction(junctionPath, targetPath));
+            DirectoryInfo junctionInfo = new(junctionPath);
+            return junctionInfo;
+        }
+
+        protected abstract DirectoryInfo CreateDirectory(string path);
+
+        protected abstract FileSystemInfo ResolveLinkTarget(string junctionPath, bool returnFinalTarget);
+
+        protected abstract void VerifyEnumerateMethods(string junctionPath, string[] expectedFiles, string[] expectedDirectories, string[] expectedEntries);
+
+        protected void VerifyEnumeration(IEnumerable<string> expectedEnumeration, IEnumerable<string> actualEnumeration)
+        {
+            foreach (string expectedItem in expectedEnumeration)
+            {
+                Assert.True(actualEnumeration.Contains(expectedItem));
+            }
```

Comment:
Collection comparison where order does not matter. If there's a simpler way of performing this (but as clear as this), let me know.